src/business-logic/services/test/device-service.spec.ts

```diff
@@ -0,0 +1,41 @@
+import {DeviceServiceImplementation} from '../device-service-implementation'
+import {Device, TelemetricsDevice} from '../../../model/entities/device'
+import {instance, mock, verify, when} from '@typestrong/ts-mockito'
+import {MongoDeviceRepository} from '../../../data-access/repositories/mongo/mongo-device-repository'
+import {DeviceService} from '../interfaces/device-service'
+import {DeviceEventEmitter} from '../interfaces/device-event-emitter'
+
+describe('DeviceService', () => {
```

Comment:
This file is called `device-service.spec.ts` but what you are testing is your `DeviceServiceImplementation`, so the file/describes should reflect that.